maint-lib/ceph_version.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+CEPH_VERSION_SPEC=$1
+WHAT_TO_EXTRACT=$2
+
+# Search for the two possible separator between CEPH_VERSION and CEPH_POINT_RELEASE
+# Let's consider CEPH_VERSION_SPEC=luminous-12.2.0-1
+for separator in "=" "-"; do
+  # If the line doesn't have a separator, let's try the next one
+  echo "${CEPH_VERSION_SPEC}" | grep -q "${separator}" || continue
+  case "${WHAT_TO_EXTRACT}" in
+    "CEPH_POINT_RELEASE")
+      # e.g Extracting -12.2.0-1 from CEPH_VERSION_SPEC
+      echo -n "${separator}"
+      echo "${CEPH_VERSION_SPEC}" | cut -d "${separator}" -f 2-
+      exit 0
+      ;;
+    *)
```

Comment:
What to extract can only be 2 values in reality 'CEPH_POINT_RELEASE' or 'CEPH_VERSION'. But in code, the values can be 'CEPH_POINT_RELEASE' or <anything> which could be problematic. These case statements should specify only the 2 possible values, and the default option `*)` should exit with error.

Reply:
That was done on purpose. We'll have CEPH_VERSION unless we explicit wants 'CEPH_POINT_RELEASE'